core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -673,7 +674,7 @@ private RexNode simplifyIs2(SqlKind kind, RexNode a, RexUnknownAs unknownAs) {
         return simplify(a, unknownAs);
       } else {
         RexNode newSub =
-            simplify(a, kind == SqlKind.IS_TRUE ? RexUnknownAs.FALSE : RexUnknownAs.TRUE);
+            simplify(a, kind == SqlKind.IS_TRUE ? RexUnknownAs.FALSE : TRUE);
```

Comment:
Can we keep it as RexUnknownAs.TRUE to make the code more readable ?

Reply:
Changed back to RexUnknownAs.TRUE